packages/react-dom-bindings/src/client/ReactDOMTextarea.js

```diff
@@ -57,20 +49,48 @@ export function initWrapperState(element: Element, props: Object) {
       );
       didWarnValDefaultVal = true;
     }
+    if (props.children != null && props.value == null) {
+      console.error(
+        'Use the `defaultValue` or `value` props instead of setting ' +
+          'children on <textarea>.',
+      );
+    }
+  }
+}
+
+export function updateTextarea(element: Element, props: Object) {
+  const node: HTMLTextAreaElement = (element: any);
+  const value = getToStringValue(props.value);
+  const defaultValue = getToStringValue(props.defaultValue);
+  if (defaultValue != null) {
+    node.defaultValue = toString(defaultValue);
+  } else {
+    node.defaultValue = '';
+  }
+  if (value != null) {
+    // Cast `value` to a string to ensure the value is set correctly. While
+    // browsers typically do this as necessary, jsdom doesn't.
+    const newValue = toString(value);
+    // To avoid side effects (such as losing text selection), only set value if changed
+    if (newValue !== node.value) {
+      node.value = newValue;
+    }
+    // TOOO: This should respect disableInputAttributeSyncing flag.
```

Comment:
@gaearon This doesn't update an attribute but it does update the internal state so it falls under the same category of issues.